src/core/task/Task.ts

```diff
@@ -1470,6 +1502,28 @@ export class Task extends EventEmitter<ClineEvents> {
 			autoCondenseContextPercent = 100,
 		} = (await this.providerRef.deref()?.getState()) ?? {}
 
+		// Get condensing configuration for automatic triggers
+		const state = await this.providerRef.deref()?.getState()
```

Comment:
Code might be a little cleaner if you put this at the top of the method `const state = await this.providerRef.deref()?.getState() ?? {}` and then just directly refer to `state.customCondensingPrompt` etc